### ISSUE TITLE
Emails get sent even if course is hidden

### ISSUE THREAD

**Issue by hribeir0** [catalyst/moodle-mod_reengagement#42]
Hi,
I've changed some ideas with Dan on the forum about the fact that the messages are sent even if the course is hidden. 
I've tweaked the SQL a bit in order to check that variable.  What do you think?

`SELECT cm.id as id, cm.id as cmid, cm.availability, r.id as rid, r.course as courseid,
                                r.duration, r.emaildelay
                          FROM {reengagement} r
                          INNER JOIN {course_modules} cm on cm.instance = r.id
			  JOIN {course} c ON c.id = r.course
                          JOIN {modules} m on m.id = cm.module
                          WHERE m.name = 'reengagement'
			  AND c.visible = 1
                      ORDER BY r.id ASC`
Thanks

**Comment by danmarsden** [comment]
HI @hribeir0 - first impression without running the sql is that it should work for Moodle - we'll need to think about how we implement it though because this plugin also supports Totara and course visibility is managed slightly differently.

**Comment by hribeir0** [comment]
I've made a couple of tests and it seems to work correctly but please have a look. 
I'm not familiarized with Totara so can't help.

**Comment by danmarsden** [comment]
yeah - that's more a question for us when/if we decide to patch this in the code - great that it's working for you.
